modules/ckanext-ytp_main/ckanext/ytp/dcat.py

```diff
@@ -115,3 +156,57 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):
             group_titles = (t for t in group_dict.get('title_translated', {}).values() if t)
             for title in group_titles:
                 g.add((theme, SKOS.prefLabel, Literal(title)))
+
+        # dcat:landingPage
+        external_urls = (u for u in dataset_dict.get('external_urls', []) if u)
+
+        for external_url in external_urls:
+            document = URIRef(external_url)
+            g.add((document, RDF.type, FOAF.Document))
+            g.add((dataset_ref, DCAT.landingPage, document))
+
+        # dct:spatial
+        geographical_coverages = set(g for g in dataset_dict.get('geographical_coverage', []) if g)
+
+        for geographical_coverage in geographical_coverages:
+            location = BNode()
+            g.add((dataset_ref, DCT.spatial, location))
+            g.add((location, RDF.type, DCT.Location))
+            g.add((location, DCAT.centroid, Literal(geographical_coverage)))
+
+        # dct:accuralPeriodicity
+        update_frequencies = set(u for lang in dataset_dict.get('update_frequency', {}).values() for u in lang if u)
+
+        for update_frequency in update_frequencies:
+            g.add((dataset_ref, DCT.accrualPeriodicity, Literal(update_frequency)))
+
+        # dct:type
+        content_types = set(t for lang in dataset_dict.get('content_type', {}).values() for t in lang if t)
+
+        for content_type in content_types:
+            concept = Literal(content_type)
+            g.add((dataset_ref, DCT.type, concept))
+
+        # dct:identifier
+        g.add((dataset_ref, DCT.identifier, Literal(dataset_dict.get('id'))))
+
+        # dct:temporal
+        valid_from = dataset_dict.get('valid_from')
+        valid_till = dataset_dict.get('valid_till')
+
+        if valid_from or valid_till:
+            period = BNode()
+            g.add((dataset_ref, DCT.temporal, period))
+            g.add((period, RDF.type, DCT.PeriodOfTime))
+            if valid_from:
+                g.add((period, DCAT.startDate, Literal(valid_from)))
+            if valid_till:
+                g.add((period, DCAT.endDate, Literal(valid_till)))
+
+        # dct:issued
+        date_released = dataset_dict.get('date_released')
+
+        if date_released:
```

Comment:
date_released is used only with 6aika datasets, the rest use ckan core field metadata_created.